src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -6036,7 +6036,18 @@ void StorageReplicatedMergeTree::waitMutation(const String & znode_name, size_t
     auto zookeeper = getZooKeeper();
     Strings replicas;
     if (mutations_sync == 2) /// wait for all replicas
+    {
         replicas = zookeeper->getChildren(fs::path(zookeeper_path) / "replicas");
+        /// This replica should be first, to ensure that the mutation will be loaded into memory
+        for (auto it = replicas.begin(); it != replicas.end(); ++it)
+        {
+            if (*it == replica_name)
+            {
+                std::iter_swap(it, replicas.rbegin());
```

Comment:
It makes this replica the last one, not first

Reply:
Indeed, nice catch! I guess that this was a typo `rbegin` vs `begin`. Anyway - #39902